apps/desktop-wallet/src/features/thumbnails/NFTThumbnail.tsx

```diff
@@ -38,71 +38,77 @@ interface NFTThumbnailProps {
-    hideIfError,
-    borderRadius,
-    autoPlay,
-    showPlayIconIfVideo,
-    playOnHover,
-    onClick
-  }: NFTThumbnailProps) => {
-    const { t } = useTranslation()
-    const theme = useTheme()
-
-    const { data: nft, error, isLoading } = useFetchNft({ id: nftId })
-
-    if (error && !hideIfError)
-      return (
-        <ErrorMessage>
-          <X />
-        </ErrorMessage>
-      )
-
-    const src = nft?.image
-
-    return (
-      <Container
-        style={{
-          borderRadius: borderRadius || 0,
-          boxShadow: border ? `0 0 0 1px ${theme.border.primary}` : 'initial',
-          height: size,
-          width: size
-        }}
-        onClick={onClick}
-      >
-        {!isLoading ? (
-          src ? (
-            nft.dataType === 'image' ? (
-              <Image src={src} height={size} width={size} />
-            ) : nft.dataType === 'video' ? (
-              autoPlay ? (
-                <video src={src} autoPlay={autoPlay} loop width="100%" height="100%" preload="auto" muted playsInline />
-              ) : (
-                <VideoThumbnail videoUrl={src} showPlayIcon={showPlayIconIfVideo} playOnHover={playOnHover} />
-              )
-            ) : (
-              <ErrorMessage>{t('Unsupported media type')}</ErrorMessage>
-            )
-          ) : (
-            <ErrorMessage>{t('No media')}</ErrorMessage>
-          )
-        ) : (
-          <SpinnerContainer>
-            <Spinner />
-          </SpinnerContainer>
-        )}
-      </Container>
+const NFTThumbnail = ({ border, borderRadius, onClick, hideIfError, ...props }: NFTThumbnailProps) => {
+  const { error } = useFetchNft({ id: props.nftId })
+
+  if (hideIfError && error) return null
+
+  return (
+    <NftThumbnailStyled border={border} borderRadius={borderRadius} onClick={onClick} size={props.size}>
+      <NftThumbnailMedia {...props} />
+    </NftThumbnailStyled>
+  )
+}
+
+export default NFTThumbnail
+
+const NftThumbnailMedia = memo(({ nftId, size, ...props }: NFTThumbnailProps) => {
+  const { t } = useTranslation()
+
+  const { data: nft, isLoading, error: fetchNftError } = useFetchNft({ id: nftId })
+
+  if (isLoading) return <SkeletonLoaderStyled height={size} />
+
+  if (fetchNftError || !nft || (nft && !nft.image)) return <MissingMediaPlaceholder />
+
+  if (nft.dataType === 'image') return <NftThumbnailMediaImage src={nft.image} alt={nft.description} size={size} />
+
+  if (nft.dataType === 'video') return <NftThumbnailMediaVideo {...props} src={nft.image} />
+
+  return <ErrorMessage>{t('Unsupported media type')}</ErrorMessage>
+})
+
+interface NftThumbnailMediaImageProps {
+  src: string
+  alt?: string
+  size?: string
+}
+
+const NftThumbnailMediaImage = memo(({ src, alt, size }: NftThumbnailMediaImageProps) => {
+  const [error, setError] = useState(false)
+
+  if (error) return <MissingMediaPlaceholder />
+
+  return <Image src={src} alt={alt} height={size} width={size} onError={() => setError(true)} />
```

Comment:
The image component itself might fail to load even if the query fetch has succeeded. We need to handle this case with the `onError` callback.